rocketpy/environment/environment.py

```diff
@@ -3142,22 +3147,26 @@ def add_wind_gust(self, wind_gust_x, wind_gust_y):
         # Reset wind heading and velocity magnitude
         self.wind_heading = Function(
             lambda h: (180 / np.pi)
-            * np.arctan2(self.wind_velocity_x(h), self.wind_velocity_y(h))
+            * np.arctan2(
+                self.wind_velocity_x.get_value_opt(h),
+                self.wind_velocity_y.get_value_opt(h),
+            )
             % 360,
             "Height (m)",
             "Wind Heading (degrees)",
             extrapolation="constant",
         )
         self.wind_speed = Function(
-            lambda h: (self.wind_velocity_x(h) ** 2 + self.wind_velocity_y(h) ** 2)
+            lambda h: (
+                self.wind_velocity_x.get_value_opt(h) ** 2
+                + self.wind_velocity_y.get_value_opt(h) ** 2
+            )
```

Comment:
TODO: Analize if we could set np.ndarray-like Functions instead of using lambdas and callable in these Environment assignments.